### PR TITLE
Remove duplicate Promise.all tests

### PR DESCRIPTION
**Pull request by marjakh**
1) Promise/all/S25.4.4.1_A6.1_T1.js is the same as
Promise/all/S25.4.4.1_A2.1_T1.js

2) Promise/all/S25.4.4.1_A6.1_T2.js is covered by
Promise/all/S25.4.4.1_A2.3_T1.js and
Promise/all/S25.4.4.1_A2.3_T2.js

### CODE REVIEW

**Comment by mathiasbynens** [comment]
cc @leobalter

File: test/built-ins/Promise/all/S25.4.4.1_A6.1_T1.js [hunk 1]
```diff
@@ -1,16 +0,0 @@
...
-    Promise.all with 0-element array
-es6id: S25.4.4.1_A5.1_T1
-author: Sam Mikes
-description: Promise.all([]) produces a promise
----*/
-
-var p = Promise.all([]);
-
-if (!(p instanceof Promise)) {
-  $ERROR('Expected Promise.all([]) to be instanceof Promise' + err);
```
Comment (leobalter) [inline test/built-ins/Promise/all/S25.4.4.1_A6.1_T1.js:15]: lol, what's even `err` here?

**Comment by leobalter** [comment]
The linter is running forever, but there isn't anything to worry about in this PR.

**Comment by leobalter** [comment]
I'd rather remove Promise/all/S25.4.4.1_A2.3_T1.js and Promise/all/S25.4.4.1_A2.3_T2.js to end with a single file, but either way is fine. These tests need an eventual refactoring for a better structure.

**Pull request merged**